src/inmanta/protocol/rest/__init__.py

```diff
@@ -469,21 +474,104 @@ async def process_return(self, config: common.UrlMethod, result: Apireturn) -> c
-                if config.properties.reply:
+                if self._config.properties.reply:
                     return common.Response.create(
                         ReturnValue(status_code=code, response=body),
-                        config.properties.envelope,
-                        config.properties.envelope_key,
+                        self._config.properties.envelope,
+                        self._config.properties.envelope_key,
                     )
                 else:
                     LOGGER.warning("Method %s returned a result although it has no reply!")
 
             return common.Response.create(ReturnValue(status_code=code, response=None), envelope=False)
 
+    def _parse_and_validate_auth_token(self) -> None:
+        """Get the auth token provided by the caller and decode it.
+
+        :return: A mapping of claims
+        """
+        header_value: Optional[str] = None
+        header_name: Optional[str] = None
+        cfg_name: Optional[str] = None
+        for name in auth.AuthJWTConfig.list():
+            cfg = auth.AuthJWTConfig.get(name)
+
+            if cfg.jwt_header in self._request_headers:
+                header_value = self._request_headers[cfg.jwt_header]
+                header_name = cfg.jwt_header
+                cfg_name = name
+
+        if not header_value or not header_name or not cfg_name:
+            return None
+
+        if " " in header_value:
+            parts = header_value.split(" ")
+
+            if len(parts) != 2 or parts[0].lower() != "bearer":
+                logging.getLogger(__name__).warning(
+                    f"Invalid JWT token header ({header_name} from auth config {cfg_name}). "
+                    f"A bearer token is expected, instead ({header_value} was provided)"
+                )
+                return None
+
+            token_value = parts[1]
+        else:
+            token_value = header_value
+
+        self._auth_token = auth.decode_token(token_value)
+
+        cfg = auth.AuthJWTConfig.get(cfg_name)
+        if cfg.jwt_username_claim in self._auth_token:
+            self._auth_username = self._auth_token[cfg.jwt_username_claim]
+
+    def authenticate(self, auth_enabled: bool) -> None:
+        """Fetch any identity information and authenticate. This will also load this authentication
+        information in this instance.
+
+        :param auth_enabled: is authentication enabled?
+        """
+        if not auth_enabled:
+            return
+
+        # get and validate the token. A valid token means that user is authenticated
+        self._parse_and_validate_auth_token()
+        if self._auth_token is None and self._config.properties.enforce_auth:
+            # We only need a valid token when the
+            raise exceptions.UnauthorizedException()
+
+    def authorize_request(self) -> None:
+        """
+        Authorize a request based on the given data
+        """
+        if self._auth_token is None:
```

Comment:
it might be safer to also raise a `UnauthorizedException()` if auth is enabled.

Reply:
done